packages/portfolio-contract/test/portfolio-open.test.ts

```diff
@@ -0,0 +1,164 @@
+    exit(completion) {
+      if (hasExited) throw Error('already exited');
+      log({ _cap: 'seat', _method: 'exit', completion });
+      hasExited = true;
+    },
+    fail(reason) {
+      if (hasExited) throw Error('already exited');
+      log({ _cap: 'seat', _method: 'fail', reason });
+      hasExited = true;
+    },
+  } as ZCFSeat;
+
+  const inertSubscriber = {} as ResolvedPublicTopic<never>['subscriber'];
+
+  return { orch, log: buf, makePortfolioKit, zoeTools, seat, inertSubscriber };
+};
+
+test('open portfolio', async t => {
+  const { orch, log, makePortfolioKit, zoeTools, seat, inertSubscriber } =
+    mocks();
+
+  const localP = makeLocalAccount(orch, {});
+  const actual = await openPortfolio(
+    orch,
+    { zoeTools, makePortfolioKit, inertSubscriber },
+    seat,
+    undefined,
+    localP,
+  );
+  t.snapshot(log, 'call log');
+  t.is(passStyleOf(actual.invitationMakers), 'remotable');
+  t.like(actual.publicTopics, [
+    { description: 'USDN ICA', storagePath: 'cosmos:noble-3:noble11028' },
+  ]);
+});
+
+test('handle failure in localTransfer from seat to local account', async t => {
+  const { orch, log, makePortfolioKit, zoeTools, seat, inertSubscriber } =
+    mocks({ localTransfer: Error('localTransfer from seat failed') });
+
+  const localP = makeLocalAccount(orch, {});
+  const actual = await openPortfolio(
+    orch,
+    { zoeTools, makePortfolioKit, inertSubscriber },
+    seat,
+    undefined,
+    localP,
+  );
+  t.snapshot(log, 'call log');
+  t.is(passStyleOf(actual.invitationMakers), 'remotable');
+  t.like(actual.publicTopics, [
+    { description: 'USDN ICA', storagePath: 'cosmos:noble-3:noble11028' },
+  ]);
+  const [{ storagePath: ICAAddr }] = actual.publicTopics;
+  t.log('we still get the ICA address', ICAAddr);
+});
+
+test('handle failure in IBC transfer', async t => {
+  const { orch, log, makePortfolioKit, zoeTools, seat, inertSubscriber } =
+    mocks({ transfer: Error('IBC transfer failed') });
+
+  const localP = makeLocalAccount(orch, {});
+  const actual = await openPortfolio(
+    orch,
+    { zoeTools, makePortfolioKit, inertSubscriber },
+    seat,
+    undefined,
+    localP,
+  );
+  t.snapshot(log, 'call log');
+  t.is(passStyleOf(actual.invitationMakers), 'remotable');
+  t.like(actual.publicTopics, [
+    { description: 'USDN ICA', storagePath: 'cosmos:noble-3:noble11028' },
+  ]);
+});
+
+test('handle failure in executeEncodedTx', async t => {
+  const { orch, log, makePortfolioKit, zoeTools, seat, inertSubscriber } =
+    mocks({ executeEncodedTx: Error('Swap or Lock failed') });
+
+  const localP = makeLocalAccount(orch, {});
+  const actual = await openPortfolio(
+    orch,
+    { zoeTools, makePortfolioKit, inertSubscriber },
+    seat,
+    undefined,
+    localP,
+  );
+  t.snapshot(log, 'call log');
+  t.is(passStyleOf(actual.invitationMakers), 'remotable');
+  t.like(actual.publicTopics, [
+    { description: 'USDN ICA', storagePath: 'cosmos:noble-3:noble11028' },
```

Comment:
_nitpick_

Declare a `const` for `'cosmos:noble-3:noble11028'`

Reply:
In a contract that would be The Way.
But in tests, we tend to use literals for expected results.